docs/sql-migration-guide-upgrade.md

```diff
@@ -43,6 +43,8 @@ displayTitle: Spark SQL Upgrading Guide
 
   - In Spark version 2.4 and earlier, if `org.apache.spark.sql.functions.udf(Any, DataType)` gets a Scala closure with primitive-type argument, the returned UDF will return null if the input values is null. Since Spark 3.0, the UDF will return the default value of the Java type if the input value is null. For example, `val f = udf((x: Int) => x, IntegerType)`, `f($"x")` will return null in Spark 2.4 and earlier if column `x` is null, and return 0 in Spark 3.0. This behavior change is introduced because Spark 3.0 is built with Scala 2.12 by default.
 
+  - Since Spark 3.0, the JDBC options `lowerBound` and `upperBound` are converted to TimestampType/DateType values in the same way as casting strings to TimestampType/DateType values. The conversion is based on Proleptic Gregorian calendar, and time zone defined by the SQL config `spark.sql.session.timeZone`. In Spark version 2.4 and earlier, the conversion is based on the hybrid calendar (Julian + Gregorian) and on default system time zone.
```

Comment:
Maybe we can later make this as a group in this migration guide. For instnace,

```
- Since Spark 3.0, Proleptic Gregorian calendar ...
  - `lowerBound` and `upperBound` blah blah .
  - blah blah ..
```

Yup, we can do it in another place.

Reply:
Maybe we will collapse the separate notes into one when we will close the umbrella ticket? Because there is a common note about Proleptic Gregorian calendar but also some nuances in each change.

Reply:
Yea, sounds good.